src/v2/views/MonsterCollectionOverlay/Migration.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useMemo, useReducer } from "react";
+import { LegacyCollectionStateQuery } from "src/v2/generated/graphql";
+import { MigrationAlert, MigrationAlertItem } from "./dialog";
+import ncgIcon from "src/v2/resources/collection/items/ncg.png";
+import ClaimCollectionRewardsOverlay from "../ClaimCollectionRewardsOverlay";
+import { placeholder, useTx } from "src/v2/utils/useTx";
+
+interface MigrationProps {
+  tip: number;
+  collectionState: NonNullable<
+    LegacyCollectionStateQuery["stateQuery"]["monsterCollectionState"]
+  >;
+  collectionSheet: LegacyCollectionStateQuery["stateQuery"]["monsterCollectionSheet"];
+  onActionTxId(txId: string): void;
+}
+
+const noop = () => {};
+
+// Slight modification of src\collection\common\utils.ts:16
+function getRemain(blocks: number) {
+  const hour = blocks / 60;
+  const days = hour / 24;
+
+  if (days >= 1) return { number: days, unit: "days" } as const;
+  if (hour >= 1) return { number: hour, unit: "hours" } as const;
+
+  return { number: blocks, unit: "blocks" } as const;
+}
+
+export default function Migration({
+  tip,
+  collectionState,
+  collectionSheet,
+  onActionTxId,
+}: MigrationProps) {
+  const [isOpen, open] = useReducer(() => true, false);
+  const isClaimable = tip >= collectionState.claimableBlockIndex;
+  const tx = useTx("migrate-monster-collection", placeholder);
+
+  const deposit = useMemo(
+    () =>
+      collectionSheet?.orderedList?.find(
+        (v) => v?.level === collectionState.level
+      ),
+    [collectionState, collectionSheet]
+  );
+  const elapsedBlocks = tip - collectionState.startedBlockIndex;
+  const elapsed = getRemain(elapsedBlocks);
+
+  return (
+    <MigrationAlert isOpen={true} onConfirm={open} isClaimable={isClaimable}>
+      <MigrationAlertItem title="Deposit amount">
+        <strong>
+          <img src={ncgIcon} />
+          {deposit}
```

Comment:
Will break if collectionSheet is unavailable, or there's no matching level in the collectionSheet.

Reply:
Maybe we should show the level if it's unavailable?